File: testprojects/java/benchmark/accuracy-test-2013-04-17/domain/indirect/violatingfrom/CallInstanceMethodIndirect_MethodMethod.java
package domain.indirect.violatingfrom;

import domain.indirect.intermediate.BackgroundService;

public class CallInstanceMethodIndirect_MethodMethod {
	private BackgroundService bckg;
	private String test;
	
	public CallInstanceMethodIndirect_MethodMethod() {
		bckg = new BackgroundService();
	}

	// Indirect access via normal method 
	public String case2(){
		test = bckg.getServiceOne().getName();
		return test;
	}
}

File: testresources/csharp/benchmark/Domain/Indirect/ViolatingFrom/CallInstanceMethodIndirect_MethodMethod.cs
namespace Domain.Indirect.ViolatingFrom
{
	public class CallInstanceMethodIndirect_MethodMethod
	{
        private Domain.Indirect.Intermediate.BackgroundService bckg;

		private string test;

		public void Case2()
        {
            bckg = new Domain.Indirect.Intermediate.BackgroundService();
			test = bckg.GetServiceOne().GetName();
		}
	}
}
